packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -103,7 +106,18 @@ export const generateMirrorCell = (p: string, rwjsPaths = getPaths()) => {
 
   const typeDefPath = path.join(mirrorDir, typeDef)
   const { name } = path.parse(p)
-  writeTemplate('templates/mirror-cell.d.ts.template', typeDefPath, { name })
+
+  const fileContents = fs.readFileSync(p, 'utf-8')
+  const gqlQueries = getGqlQueries(fileContents)
+
+  // For mirror cells we only care about the first gqlQuery
+  const gqlDoc = parseGqlQueryToAst(gqlQueries[0])[0]
```

Comment:
@dac09 @jtoar I think I see a bit of a gap in the api-design of these functions here.

It's "sort of true" that we care about the first GQL query, but what we really care about is the GraphQL that has a named export with the identifier of "QUERY".

I'm happy for this to go in, but I think I would like to take a stab at making the api a bit more robust afterwards.

Reply:
You're right, I'll see what I can do!

Reply:
@peterp updated PR now... took me a while to find a good way to do this, but I think this implementation is a lot more solid. Babel with types is really hard man!

Reply:
The wallaby really helps here!

Reply:
This is awesome btw, thanks for doing it!

Reply:
Thank you for showing me how to use it, I introduced it to Dom the other day too. I think I just needed that little nudge to start using Wallaby properly, been and absolutely godsend for stuff like this